### PR TITLE
Updating Bootcamp for batch 5

### LINKED ISSUE #33: Bootcamp curriculum discussion
We'll use this issue to discuss changes to the [bootcamp curriculum](https://github.com/LDSSA/curriculum-development/blob/master/curriculum/01-bootcamp-and-binary-classification.md).

I'm proposing to remove the SLUs about ML models (other than Linear and Logistic Regression) and include them as optional content in Specializations:
- SVMs would go to Text Classification
- kNN would go to RecSys
- Tree Based models would still be in the Bootcamp (but as optional content).

Optional content means that:
- we won't present it in the bootcamp, i.e, students should go over it by themselves
- the exercises are not mandatory
- students are encouraged to study this content and they can ask questions about it 

The reasons for this proposal are:
- the bootcamp is very heavy
- if we present the models together with a more real-world example of its usage, it's more likely that the students will understand it
- by making it optional, we leave it up to the students to decide if they have the capacity, at that moment, to absorb that information, or if they prefer to do it later

### PR DESCRIPTION
**Pull request by dominguezmaria**
Important changes are:
- Changing SVM and kNN SLUs to be optional
- Present evaluation metrics for regression and classification right after introducing each of these task types
- Re-arranging topics about overfitting on the SLUs `Model Selection & Overfitting` and `Data Sufficiency & Selection` (which was renamed to `Model complexity & Overfitting`)
- Adding over and under sampling
- Including [scikit's dummy estimators](https://scikit-learn.org/stable/modules/classes.html#module-sklearn.dummy) in the `Workflow` SLU

### CODE REVIEW

File: curriculum/01-bootcamp-and-binary-classification.md [hunk 1]
```diff
@@ -9,28 +9,27 @@
...
+| SLU12 | Feature Engineering (aka Real Wold Data) | Data Skills & Analysis | Existing | [Content](#slu11) | Bootcamp Day 2 | Yes      |
+| SLU13 | Model Selection & Overfitting            | ML Fundamentals        | To Adapt | [Content](#slu12) | Bootcamp Day 2 | Yes      |
+| SLU14 | Data Sufficiency & Selection             | Data Skills & Analysis | To Adapt | [Content](#slu13) | Bootcamp Day 2 | Yes      |
+| SLU15 | Ethics & Fairness                        | Data Skills & Analysis | Existing | [Content](#slu14) | Bootcamp Day 2 | Yes      |
+| SLU16 | Hyperparameter Tuning                    | ML Fundamentals        | Existing | [Content](#slu15) | Bootcamp Day 2 | Yes      |
+| SLU17 | Workflow                                 | ML Fundamentals        | To Adapt | [Content](#slu16) | Bootcamp Day 2 | Yes      |
+| SLU18 | Support Vector Machines (SVM)            | Learning Algorithms    | New      | [Content](#slu17) | Not presented  | No       |
+| SLU19 | k-Nearest Neighbors (kNN)                | Learning Algorithms    | New      | [Content](#slu18) | Not presented  | No       |
```
Comment (CatarinaSilva) [inline curriculum/01-bootcamp-and-binary-classification.md]: The status here is not "New" right? Probably "To Adapt" ?

File: curriculum/01-bootcamp-and-binary-classification.md [hunk 2]
```diff
@@ -9,28 +9,27 @@
...
+| SLU08 | Metrics for Regression                   | ML Fundamentals        | Existing | [Content](#slu10) | Bootcamp Day 1 | Yes      |
+| SLU09 | Classification with Logistic Regression  | Learning Algorithms    | Existing | [Content](#slu08) | Bootcamp Day 1 | Yes      |
+| SLU10 | Metrics for Classification               | ML Fundamentals        | Existing | [Content](#slu09) | Bootcamp Day 1 | Yes      |
+| SLU11 | Tree-Based Models                        | Learning Algorithms    | Existing | [Content](#slu10) | Bootcamp Day 2 | Yes      |
+| SLU12 | Feature Engineering (aka Real Wold Data) | Data Skills & Analysis | Existing | [Content](#slu11) | Bootcamp Day 2 | Yes      |
+| SLU13 | Model Selection & Overfitting            | ML Fundamentals        | To Adapt | [Content](#slu12) | Bootcamp Day 2 | Yes      |
+| SLU14 | Data Sufficiency & Selection             | Data Skills & Analysis | To Adapt | [Content](#slu13) | Bootcamp Day 2 | Yes      |
+| SLU15 | Ethics & Fairness                        | Data Skills & Analysis | Existing | [Content](#slu14) | Bootcamp Day 2 | Yes      |
```
Comment (CatarinaSilva) [inline curriculum/01-bootcamp-and-binary-classification.md]: should we move ethics & fairness to the end? It doesn't have a strong practical component and it is a nice closing topic

Comment (dominguezmaria) [reply curriculum/01-bootcamp-and-binary-classification.md]: Yup, sounds good! Moved to last (presented) SLU

File: curriculum/01-bootcamp-and-binary-classification.md (hunk 1 above)
Comment (dominguezmaria) [reply curriculum/01-bootcamp-and-binary-classification.md]: 👍 fixed

**Comment by dominguezmaria** [comment]
@CatarinaSilva addressed your comments!

**Review by PedroGFonseca — APPROVED** [review]

**Pull request merged**